lib/internal/Magento/Framework/Encryption/Helper/Security.php

```diff
@@ -19,12 +15,14 @@ class Security
     /**
      * Compare two strings in a secure way that avoids string length guessing based on duration of calculation
      *
-     * @param  string $expected
-     * @param  string $actual
+     * @param string $expected
+     * @param string $actual
      * @return bool
      */
     public static function compareStrings($expected, $actual)
     {
-        return Utils::compareStrings($expected, $actual);
+        $expected     = (string) $expected;
+        $actual       = (string) $actual;
+        return hash_equals($expected, $actual);
```

Comment:
@texboy I would simplify this even further and avoid introducing one-time use variables
```suggestion
        return hash_equals((string) $expected, (string) $actual);
```